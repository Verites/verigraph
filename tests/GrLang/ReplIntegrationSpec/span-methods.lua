GrLang.add_node_type('N')
GrLang.add_edge_type('E', 'N', 'N')

X = Graph [[ a b c : N ]] 
Y = Graph [[ d e : N ]]
Z = Graph [[ ad ae be: N]]
f, g = Morphism(Z,X)[[ad ae->a; be->b]], Morphism(Z,Y)[[ad->d; ae be->e]]

s = Span(f,g)

print('span')
print(s)
print()

print('dot')
print(s:to_dot())

L1 = Graph [[ a b : N ]]
L2 = Graph [[ c d : N ]]

A, B, C, AB = Graph [[ ac : N ]], Graph [[ bd : N ]], Graph [[ bc : N ]], Graph [[ ac bd : N ]]
a = Span(Morphism(A,L1)[[ac -> a]], Morphism(A,L2)[[ac -> c]])
b = Span(Morphism(B,L1)[[bd -> b]], Morphism(B,L2)[[bd -> d]])
c = Span(Morphism(C,L1)[[bc -> b]], Morphism(C,L2)[[bc -> c]])
ab = Span(Morphism(AB,L1)[[ac -> a; bd -> b]], Morphism(AB,L2)[[ac -> c; bd -> d]])

print()
print('subobject_intersection')
print(Span.subobject_intersection(a,b))
print(Span.subobject_intersection(a,ab))

print()
print('subobject_union')
print(Span.subobject_union(a,b))
print(Span.subobject_union(a,c))
