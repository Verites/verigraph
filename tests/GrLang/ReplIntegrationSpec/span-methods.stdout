span
graph  {
  ad ae be : N
} morphism  : dom -> cod {
  ae ad -> a
  be -> b
} graph  {
  a b c : N
} morphism  : dom -> cod {
  ad -> d
  be ae -> e
} graph  {
  d e : N
}

dot
digraph  {
  subgraph cluster_Dom_ {
    graph [label=Dom_];
    node [shape=box];
    Dom_0 [label="ad:N"];
    Dom_1 [label="ae:N"];
    Dom_2 [label="be:N"]
  };
  subgraph cluster_CodL_ {
    graph [label=CodL_];
    node [shape=box];
    CodL_0 [label="a:N"];
    CodL_1 [label="b:N"];
    CodL_2 [label="c:N"]
  };
  subgraph cluster_CodR_ {
    graph [label=CodR_];
    node [shape=box];
    CodR_0 [label="d:N"];
    CodR_1 [label="e:N"]
  };
  Dom_0 -> CodL_0 [style=dotted];
  Dom_1 -> CodL_0 [style=dotted];
  Dom_2 -> CodL_1 [style=dotted];
  Dom_0 -> CodR_0 [style=dotted];
  Dom_1 -> CodR_1 [style=dotted];
  Dom_2 -> CodR_1 [style=dotted]
}

subobject_intersection
graph  {
  
} morphism  : dom -> cod {
  
} graph  {
  a b : N
} morphism  : dom -> cod {
  
} graph  {
  c d : N
}
graph  {
  ?0 : N
} morphism  : dom -> cod {
  ?0 -> a
} graph  {
  a b : N
} morphism  : dom -> cod {
  ?0 -> c
} graph  {
  c d : N
}

subobject_union
graph  {
  ac bd : N
} morphism  : dom -> cod {
  ac -> a
  bd -> b
} graph  {
  a b : N
} morphism  : dom -> cod {
  ac -> c
  bd -> d
} graph  {
  c d : N
}
nil
